configure

```diff
@@ -1027,6 +1031,91 @@ else
     echo "Checking for __builtin_ctzll ... No." | tee -a configure.log
 fi
 
+check_avx512vnni_intrinsics() {
+    # Check whether compiler supports AVX512-VNNI intrinsics
+    cat > $test.c << EOF
+"#include <stdio.h>
+#include <immintrin.h>
+int main(void) {
+    __m512i x = _mm512_set1_epi8(2);
+    const __m512i y = _mm512_set_epi8(1, 2, 3, 4, 5, 6,
+                                      7, 8, 9, 10, 11, 12, 13,
+                                      14, 15, 16, 17, 18, 19,
+                                      20, 21, 22, 23, 24, 25,
+                                      26, 27, 28, 29, 30, 31,
+                                      32, 33, 34, 35, 36, 37,
+                                      38, 39, 40, 41, 42, 43,
+                                      44, 45, 46, 47, 48, 49,
+                                      50, 51, 52, 53, 54, 55,
+                                      56, 57, 58, 59, 60, 61,
+                                      62, 63, 64);
+    __m512i z = _mm512_setzero_epi32();
+    z = _mm512_dpbusd_epi32(z, x, y);
+    (void)z;
+    return 0;
+}"
+EOF
+    if try ${CC} ${CFLAGS} ${avx512vnniflag} $test.c; then
+        echo "Checking for AVX512VNNI intrinsics ... Yes." | tee -a configure.log
+        HAVE_AVX512VNNI_INTRIN=1
+    else
+        echo "Checking for AVX512VNNI intrinsics ... No." | tee -a configure.log
+        HAVE_AVX512VNNI_INTRIN=0
+    fi
+}
+
+check_avx512_intrinsics() {
+    # Check whether compiler supports AVX512 intrinsics
+    cat > $test.c << EOF
+#include <immintrin.h>
+int main(void) {
+    __m512i x = _mm512_set1_epi8(2);
+    const __m512i y = _mm512_set_epi8(1, 2, 3, 4, 5, 6,
+                                      7, 8, 9, 10, 11, 12, 13,
+                                      14, 15, 16, 17, 18, 19,
+                                      20, 21, 22, 23, 24, 25,
+                                      26, 27, 28, 29, 30, 31,
+                                      32, 33, 34, 35, 36, 37,
+                                      38, 39, 40, 41, 42, 43,
+                                      44, 45, 46, 47, 48, 49,
+                                      50, 51, 52, 53, 54, 55,
+                                      56, 57, 58, 59, 60, 61,
+                                      62, 63, 64);
+    x = _mm512_sub_epi8(x, y);
+    (void)x;
+    return 0;
+}
+EOF
+    if try ${CC} ${CFLAGS} ${avx512flag} $test.c; then
+        echo "Checking for AVX512 intrinsics ... Yes." | tee -a configure.log
+        HAVE_AVX512_INTRIN=1
+    else
+        echo "Checking for AVX512 intrinsics ... No." | tee -a configure.log
+        HAVE_AVX512_INTRIN=0
+    fi
+}
+
+check_mask_intrinsics() {
+    # Check whether compiler supports AVX512 k-mask intrinsics
+    cat > $test.c << EOF
+#include <immintrin.h>
+#include <x86intrin.h>
```

Comment:
Do we need x86intrin.h here?

Reply:
Don't think so - not sure how that snuck in.  There were at any given point a lot of missing intrinsics amongst all the compilers (possibly even ICC's).  It's hard to get that bit straight.